### DISCUSSION TITLE
Intuición detrás de Kosaraju

### DISCUSSION THREAD

**Discussion by sofiacorreabusquets** [IIC2133-PUC/2024-1#184] [Preguntas materia] [answered]
Hola :)

He estado estudiando el algoritmo de Kosaraju y cada vez que veo el código o un video del algoritmo ejecutándose logro entenderlo, pero no me entra la intuición detrás de por qué funciona para encontrar todas las CFCs. Hay algún ejercicio visual, video, analogía o forma de pensarlo que me puedan recomendar para entender su correctitud?

Gracias de antemano!

**Comment by Vinuelax** [comment, marked as answer]
Hola,
te dejo un link que a mí me ayudo a comprender el algoritmo y tiene una buena idea de como generar la intuición de la correctitud.
[Artículo](https://www.baeldung.com/cs/kosaraju-algorithm-scc)
[Ejercicio paso a paso](https://www.youtube.com/watch?v=5wFyZJ8yH9Q&ab_channel=GeeksforGeeks)

Si aún no te hace sentido, después puedo ver si encuentro/creo más material.
